gordon/main.py

```diff
@@ -83,14 +83,33 @@ def setup(config_root=''):
     return config
 
 
+def _log_or_raise_exceptions(errors, debug):
+    log_level_func = logging.warn
+    if not debug:
+        log_level_func = logging.error
+
+    base_msg = 'Plugin "{name}" was not loaded:'
+    for name, exc in errors:
+        msg = base_msg.format(
+            name=name, etype=exc.__class__.__name__, emsg=exc)
+        log_level_func(msg, exc_info=exc)
+
+    if not debug:
+        raise SystemExit(1)
```

Comment:
What does this help test? I don't understand the point of running if plugins are missing. Won't Gordon just throw errors when it tries to access any of the plugin's features

Reply:
So, I'm coming from the PoV of potentially trying to debug more than one plugin. Multiple ones could be failing, and/or conflicting with each other. And rather than just removing one at a time within config (i.e. `core.plugins = ["some", "plugins"]`) to try and figure it out (and potentially different behavior if plugins are conflicting), this could give a fuller picture.